src/components/UserSelector.tsx

```diff
@@ -1,24 +1,40 @@
-import React, { useContext, useEffect, useState } from 'react';
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useEffect, useState } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
 import { User } from '../types/User';
+import { useAppSelector } from '../hooks/useAppSelector.hook';
+import { fetchAllUsers } from '../slices/users.slice';
+import { useAppDispatch } from '../hooks/useAppDispatch.hook';
+import { setAuthor } from '../slices/author.slice';
 
 type Props = {
   value: User | null;
-  onChange: (user: User) => void;
 };
 
 export const UserSelector: React.FC<Props> = ({
   // `value` and `onChange` are traditional names for the form field
   // `selectedUser` represents what actually stored here
   value: selectedUser,
-  onChange,
 }) => {
   // `users` are loaded from the API, so for the performance reasons
   // we load them once in the `UsersContext` when the `App` is opened
   // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
+  // const users = useContext(UserContext);
   const [expanded, setExpanded] = useState(false);
+  const dispatch = useAppDispatch();
+  const { users } = useAppSelector(state => state.users);
+
+  const fetchUsers = async () => {
+    await dispatch(fetchAllUsers());
+  };
+
+  const handleClickChangeUser = (newUser: User) => {
+    dispatch(setAuthor(newUser));
+  };
+
+  useEffect(() => {
+    fetchUsers();
+  }, []);
```

Comment:
The `useEffect` dependency array is empty, which means `fetchUsers` will only be called once when the component mounts. This is correct if you intend to fetch users only once. If you need to refetch users under certain conditions, consider adding dependencies.